core/iwasm/compilation/aot_llvm_extra.cpp

```diff
@@ -381,28 +411,53 @@ wasm_dpgo_set_function_entry_count(LLVMValueRef function, uint32 count)
 
 /* append !{!"branch_weights", i32 NN, i32 MM} */
 void
-wasm_dpgo_set_branch_weights(LLVMModuleRef module, LLVMValueRef instruction,
+wasm_dpgo_set_branch_weights(LLVMContextRef context, LLVMValueRef instruction,
                              uint32 *counts, uint32 counts_size)
 {
-    Module *M = reinterpret_cast<Module *>(module);
-    MDBuilder MDB(M->getContext());
-
+    MDBuilder MDB(*reinterpret_cast<LLVMContext *>(context));
     SmallVector<unsigned, 4> Weights;
-    for (unsigned i = 0; i < counts_size; i++)
+
+    for (unsigned i = 0; i < counts_size; i++) {
         Weights.push_back(counts[i]);
+    }
 
+    // FIXME: add more counters when meeting switch
+    LOG_DEBUG("%s --> branch_weights {%u, %u}",
+              LLVMPrintValueToString(instruction), counts[0], counts[1]);
+
+    bh_assert(LLVMGetValueKind(instruction) == LLVMInstructionValueKind);
     Instruction *I = unwrap<Instruction>(instruction);
     I->setMetadata(LLVMContext::MD_prof, MDB.createBranchWeights(Weights));
 }
 
+void
+wasm_dpgo_unlike_true_branch(LLVMContextRef context, LLVMValueRef cond_br)
+{
+    bh_assert(LLVMIsABranchInst(cond_br));
+    bh_assert(LLVMIsConditional(cond_br));
+
+    uint32 weights[2] = { 1, 1000 };
```

Comment:
We will update the values after profiling, right?

Reply:
Actually no. both `wasm_dpgo_unlike_true_branch()` and `wasm_dpgo_unlike_false_branch()` kinds similar with `unlikely()` and `likely()`. The programmer is very sure about the most likely path during coding.

We are using both functions for exception handling. We will give the exception handler target a very low weight, usually 1, and the normal path a high weight, usually 1000.

Since those exception handling llvm jit generated is different from fast jit generated, there is no counter for those *condbr* in exception handling. But lucky for us, we don't need that.